src/esp/sim/Simulator.cpp

```diff
@@ -1170,6 +1197,10 @@ bool Simulator::drawObservation(const int agentId,
   return false;
 }
 
+bool Simulator::drawObservation(esp::sensor::VisualSensor* visualSensor) {
```

Comment:
This should be a reference.  Pointer implies it's a nullable or the called method will change what it points to.